content/existential-containers.md

```diff
@@ -0,0 +1,206 @@
+The companion object of `Containing` provides basic support to create containers ergonomically.
+For instance:
+
+```scala
+def largest(xs: Seq[Containing[Polygon]]): Option[Containing[Polygon]] =
+  xs.maxByOption(_.value.area)
+```
+
+To further improve usability, we propose to let the compiler inject the selection of the `value` field implicitly when a method of `Containing[C]` is selected.
+That way, one can simply write `xs.maxByOption(_.area)` in the above example, resulting in quite idiomatic scala.
+
+```scala
+// Version with subtyping:
+trait Polygon1:
+  def area: Double
+def largest1(xs: Seq[Polygon1]): Option[Polygon1] =
+  xs.maxByOption(_.area)
+
+// Version with existential containers:
+trait Polygon2 extends TypeClass:
+  extension (self: Self) def area: Double
+def largest2(xs: Seq[Containing[Polygon2]]): Option[Containing[Polygon2]] =
+  xs.maxByOption(_.area)
+```
+
+### Compatibility
+
+The change in the syntax does not affect any existing code and therefore this proposal has no impact on source compatibility.
+
+The semantics of the proposed feature is fully expressible in Scala.
+Save for the implicit addition of `.value` on method selection when the receiver is an instance of `Containing[C]`, this proposal requires no change in the language.
+As a result, it has no backward binary or TASTy compatibility consequences.
+
+### Feature interactions
+
+The proposed feature is meant to interact with implicit search, as currently implemented by the language.
+More specifically, given an existential container `c`, accessing `c.value` _opens_ the existential while retaining its type `c.Value`, effectively keeping an _anchor_ (i.e., the path to the scope of the witness) to the interface of the type class.
+
+Since no change in implicit resolution is needed, this proposal cannot create unforeseen negative interactions with existing features.
+
+### Other concerns
+
+This document has been written under the experimental modularity improvements for Scala 3.
+Although the proposed feature is fully expressible without those changes, the encoding of existential containers can only work with the "old" (i.e., the one currently used in production) or "new" type class style.
+
+### Open questions
+
+One problem not addressed by the proposed encoding is the support of multiple type classes to form the interface of a specific container.
+For example, one may desire to create a container of values whose types conform to both `Polygon` _and_ `Show`.
+We have explored possible encodings for such a feature but decided to remove them from this proposal, as support for multiple type classes can most likely be achieved without any additional language change.
+
+Another open question relates to possible language support for shortening the expression of a container type and/or value.
+
+## Related work
```

Comment:
Apart from Swift, what other languages have a feature like this? Does Haskell have something similar? Or Rust? Both of those languages have typeclass-like features and presumably would encounter the same issues around e.g. hetoregenous collections. How do they solve it, or do they not?